osu.Server.Spectator.Tests/MultiplayerFlowTests.cs

```diff
@@ -510,6 +510,15 @@ public TestMultiplayerHub(MemoryDistributedCache cache)
             {
             }
 
+            protected override Task<MultiplayerRoom> RetrieveRoom(long roomId)
```

Comment:
This is bypassed here, but the rest of the methods still fail unfortunately. For example:

```
osu.Server.Spectator.Tests.MultiplayerFlowTests.AllUsersBackingOutFromLoadCancelsTransitionToPlay

MySqlConnector.MySqlException: Unable to connect to any of the specified MySQL hosts.

MySqlConnector.MySqlException
Unable to connect to any of the specified MySQL hosts.
   at MySqlConnector.Core.ServerSession.ConnectAsync(ConnectionSettings cs, Int32 startTickCount, ILoadBalancer loadBalancer, IOBehavior ioBehavior, CancellationToken cancellationToken) in /_/src/MySqlConnector/Core/ServerSession.cs:line 381
   at MySqlConnector.Core.ConnectionPool.GetSessionAsync(MySqlConnection connection, Int32 startTickCount, IOBehavior ioBehavior, CancellationToken cancellationToken) in /_/src/MySqlConnector/Core/ConnectionPool.cs:line 110
   at MySqlConnector.Core.ConnectionPool.GetSessionAsync(MySqlConnection connection, Int32 startTickCount, IOBehavior ioBehavior, CancellationToken cancellationToken) in /_/src/MySqlConnector/Core/ConnectionPool.cs:line 140
   at MySqlConnector.MySqlConnection.CreateSessionAsync(ConnectionPool pool, Int32 startTickCount, Nullable`1 ioBehavior, CancellationToken cancellationToken) in /_/src/MySqlConnector/MySqlConnection.cs:line 846
   at MySqlConnector.MySqlConnection.OpenAsync(Nullable`1 ioBehavior, CancellationToken cancellationToken) in /_/src/MySqlConnector/MySqlConnection.cs:line 410
   at MySqlConnector.MySqlConnection.Open() in /_/src/MySqlConnector/MySqlConnection.cs:line 376
   at osu.Server.Spectator.Database.GetConnection() in /home/smgi/Repos/osu-server-spectator/osu.Server.Spectator/Database.cs:line 17
   at osu.Server.Spectator.Hubs.MultiplayerHub.updateDatabaseParticipants(MultiplayerRoom room) in /home/smgi/Repos/osu-server-spectator/osu.Server.Spectator/Hubs/MultiplayerHub.cs:line 320
   at osu.Server.Spectator.Hubs.MultiplayerHub.JoinRoom(Int64 roomId) in /home/smgi/Repos/osu-server-spectator/osu.Server.Spectator/Hubs/MultiplayerHub.cs:line 85
```

Seems like perhaps the rest of the code needs to be delegated?

Reply:
Seems like we need CI on this project..

Reply:
I've fixed this and checked tests pass locally. Will add CI in a separate PR.